www/getChallenge.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+use SimpleSAML\Module\perun\DatabaseConnector;
+use SimpleSAML\Logger;
+
+$entityBody = file_get_contents('php://input');
+$body = json_decode($entityBody, true);
+
+if ($body === false) {
+    Logger::error('Perun.getChallenge: Received invalid json.');
+    http_response_code(400);
+    die;
+}
+
+if (empty($body['id'] || strlen($body['id']) < 20 || !ctype_print($body['id']))) {
+    Logger::error('Perun.getChallenge: Invalid id');
+    http_response_code(400);
+    die;
+}
+
+if (empty($body['scriptName'] || strlen($body['scriptName']) < 30 || !ctype_print($body['scriptName']))) {
+    http_response_code(400);
+    Logger::error('Perun.getChallenge: Invalid scriptName');
+    die;
+}
+
+$id = $body['id'];
+$scriptName = $body['scriptName'];
+
+const RANDOM_BYTES_LENGTH = 10;
```

Comment:
Should be at least 16 bytes (you can do up to 32 bytes because of sha256, or 64 bytes if you switch to sha512).

e.g. from WebAuthn standard:
```
In order to prevent replay attacks, the challenges MUST contain enough entropy to make guessing them infeasible. Challenges SHOULD therefore be at least 16 bytes long.
```